src/gurt/telemetry.c

```diff
@@ -2682,14 +2709,14 @@ attach_path_segment(key_t key, char *path)
 
 	/* Add a link to the new region */
 	rc = add_metric(ctx, &link_node, D_TM_LINK, NULL, NULL, path);
-	if (rc != 0) {
+	if (unlikely(rc != 0)) {
 		D_ERROR("can't set up the link node, " DF_RC "\n", DP_RC(rc));
-		D_GOTO(fail_tracking, rc);
+		D_GOTO(fail_unlock, rc);
 	}
 
 	/* track attached regions within the parent shmem */
 	parent_shmem = get_shmem_for_key(ctx, link_node->dtn_shmem_key);
-	if (parent_shmem == NULL) {
+	if (unlikely(parent_shmem == NULL)) {
 		D_ERROR("failed to get parent shmem pointer\n");
 		D_GOTO(fail_link, rc = -DER_NO_SHMEM);
 	}
```

Comment:
A little confused by this ordering on a re-read. Is this how it always was? Maybe it would be best to open the shmem first, then create the link node.

Reply:
Yes, this is the same ordering as before: https://github.com/daos-stack/daos/blob/master/src/gurt/telemetry.c#L2518